### PR TITLE
Feature: add campaign shortcodes

### PR DESCRIPTION
**Pull request by JoshuaHungDinh**
**Resolves:** [GIVE-2391], [GIVE-2392]

---

## ✨ Description

This PR introduces shortcode support for the **Campaign** and **Campaign Grid** blocks, allowing users to render them via WordPress shortcodes.

#### 🔹 `[givewp_campaign_grid]`

Renders a grid of campaigns with various layout and filtering options.

**Available Attributes:**

```php
$atts = shortcode_atts([
    'layout'           => 'full',
    'show_image'       => true,
    'show_description' => true,
    'show_goal'        => true,
    'sort_by'          => 'date',
    'order_by'         => 'desc',
    'per_page'         => 6,
    'show_pagination'  => true,
    'filter_by'        => null,
], $atts, 'givewp_campaign_grid');
```

**Example usage:**

```text
[givewp_campaign_grid layout="full" show_image="true" show_description="false" sort_by="date" order_by="asc" per_page="9" show_pagination="true"]
```

---

#### 🔹 `[givewp_campaign]`

Renders a single campaign block.

**Available Attributes:**

```php
$atts = shortcode_atts([
    'campaign_id'      => '',
    'show_image'       => true,
    'show_description' => true,
    'show_goal'        => true,
], $atts, 'givewp_campaign');
```

**Example usage:**

```text
[givewp_campaign campaign_id="33" show_image="true" show_description="false" show_goal="true"]
```

---

## 🧩 Affects

- Shortcode registration and rendering logic for the Campaign and CampaignGrid blocks.
- Public rendering of campaigns via `[givewp_campaign]` and `[givewp_campaign_grid]`.

---

## 🖼️ Visuals

---

## ✅ Testing Instructions

1. Add the shortcodes above to a page or post.
2. Confirm assets are loaded correctly (`JS`, `CSS`).
3. Ensure output matches expected layout and data.
4. Test edge cases (e.g., invalid `campaign_id`, zero results in grid).

---

## 🔍 Pre-review Checklist

- [x] Acceptance criteria satisfied and marked in related issues.
- [x] Relevant `@unreleased` tags included in DocBlocks.
- [ ] Includes unit tests where applicable.
- [ ] Reviewed by the designer (if visual changes apply).
- [x] [[Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review)](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed.

[GIVE-2391]: https://stellarwp.atlassian.net/browse/GIVE-2391?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[GIVE-2392]: https://stellarwp.atlassian.net/browse/GIVE-2392?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Campaigns/Blocks/Campaign/CampaignShortcode.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+     * @return string
+     */
+    public function renderShortcode($atts): string
+    {
+        $this->loadAssets();
+        $attributes = $this->parseAttributes($atts);
+
+        $renderFile = GIVE_PLUGIN_DIR . 'src/Campaigns/Blocks/Campaign/render.php';
+
+        if (file_exists($renderFile)) {
```
Comment (alaca) [inline src/Campaigns/Blocks/Campaign/CampaignShortcode.php:24]: Do we need this check? We already know this file exists.

**Review by alaca — APPROVED** [review]
Good job, @JoshuaHungDinh

File: src/Campaigns/Blocks/Campaign/CampaignShortcode.php (hunk 1 above)
Comment (JoshuaHungDinh) [reply src/Campaigns/Blocks/Campaign/CampaignShortcode.php]: I suppose not. Resolved: 507964f14597b0a48f969b7b4eb6c98756c69756

**Review by rickalday — APPROVED** [review]
Passed QA tests
